Bottom: 71de9a1b9ed11cb45b4c82812729122ca008bd94
Top:    2d0667281bc1b8fe6081ef50347bae23f22551cf
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-09-05 13:00:47 +0100

hid/gtk: Use gtk_widget_get_window() accessor

In GTK3.0, direct access to widget->window will be impossible.


---

diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index 1eed303..6df553e 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -261,7 +261,8 @@ ghid_shift_is_pressed ()
   if( ! ghid_gui_is_up ) 
     return 0;
 
-  gdk_window_get_pointer (out->drawing_area->window, NULL, NULL, &mask);
+  gdk_window_get_pointer (gtk_widget_get_window (out->drawing_area),
+                          NULL, NULL, &mask);
   return (mask & GDK_SHIFT_MASK) ? TRUE : FALSE;
 }
 
@@ -274,7 +275,8 @@ ghid_control_is_pressed ()
   if( ! ghid_gui_is_up )
     return 0;
 
-  gdk_window_get_pointer (out->drawing_area->window, NULL, NULL, &mask);
+  gdk_window_get_pointer (gtk_widget_get_window (out->drawing_area),
+                          NULL, NULL, &mask);
   return (mask & GDK_CONTROL_MASK) ? TRUE : FALSE;
 }
 
@@ -287,7 +289,8 @@ ghid_mod1_is_pressed ()
   if( ! ghid_gui_is_up )
     return 0;
 
-  gdk_window_get_pointer (out->drawing_area->window, NULL, NULL, &mask);
+  gdk_window_get_pointer (gtk_widget_get_window (out->drawing_area),
+                          NULL, NULL, &mask);
 #ifdef __APPLE__
   return (mask & ( 1 << 13 ) ) ? TRUE : FALSE;  // The option key is not MOD1, although it should be...
 #else
@@ -326,7 +329,8 @@ ghid_set_crosshair (int x, int y, int action)
    * and gdk_display_warp_pointer work relative to the whole display, whilst
    * our coordinates are relative to the drawing area origin.
    */
-  gdk_window_get_origin (gport->drawing_area->window, &offset_x, &offset_y);
+  gdk_window_get_origin (gtk_widget_get_window (gport->drawing_area),
+                         &offset_x, &offset_y);
   display = gdk_display_get_default ();
 
   switch (action) {
@@ -1544,7 +1548,8 @@ Benchmark (int argc, char **argv, Coord x, Coord y)
   do
     {
       ghid_invalidate_all ();
-      gdk_window_process_updates (gport->drawing_area->window, FALSE);
+      gdk_window_process_updates (gtk_widget_get_window (gport->drawing_area),
+                                  FALSE);
       time (&end);
       i++;
     }
@@ -1595,7 +1600,8 @@ Center(int argc, char **argv, Coord pcb_x, Coord pcb_y)
    */
 
   ghid_pcb_to_event_coords (pcb_x, pcb_y, &widget_x, &widget_y);
-  gdk_window_get_origin (gport->drawing_area->window, &offset_x, &offset_y);
+  gdk_window_get_origin (gtk_widget_get_window (gport->drawing_area),
+                         &offset_x, &offset_y);
 
   pointer_x = offset_x + widget_x;
   pointer_y = offset_y + widget_y;
diff --git a/src/hid/gtk/gui-misc.c b/src/hid/gtk/gui-misc.c
index dbed88e..0f8daf7 100644
--- a/src/hid/gtk/gui-misc.c
+++ b/src/hid/gtk/gui-misc.c
@@ -382,7 +382,8 @@ ghid_get_pointer (int *x, int *y)
 {
   gint xp, yp;
 
-  gdk_window_get_pointer (gport->drawing_area->window, &xp, &yp, NULL);
+  gdk_window_get_pointer (gtk_widget_get_window (gport->drawing_area),
+                          &xp, &yp, NULL);
   if (x)
     *x = xp;
   if (y)
diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index b9c5ab3..22dd54e 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -121,7 +121,7 @@ ghid_note_event_location (GdkEventButton * ev)
 
   if (!ev)
     {
-      gdk_window_get_pointer (ghid_port.drawing_area->window,
+      gdk_window_get_pointer (gtk_widget_get_window (ghid_port.drawing_area),
                               &event_x, &event_y, NULL);
     }
   else
@@ -391,7 +391,7 @@ ghid_port_drawing_area_configure_event_cb (GtkWidget * widget,
   if (gport->pixmap)
     gdk_pixmap_unref (gport->pixmap);
 
-  gport->pixmap = gdk_pixmap_new (widget->window,
+  gport->pixmap = gdk_pixmap_new (gtk_widget_get_window (widget),
 				  gport->width, gport->height, -1);
   gport->drawable = gport->pixmap;
 
diff --git a/src/hid/gtk/gui-pinout-preview.c b/src/hid/gtk/gui-pinout-preview.c
index 88576a1..c4bef20 100644
--- a/src/hid/gtk/gui-pinout-preview.c
+++ b/src/hid/gtk/gui-pinout-preview.c
@@ -197,7 +197,8 @@ ghid_pinout_preview_set_property (GObject * object, guint property_id,
     case PROP_ELEMENT_DATA:
       pinout_set_data (pinout, (ElementType *)g_value_get_pointer (value));
       if (GTK_WIDGET_REALIZED (GTK_WIDGET (pinout)))
-	gdk_window_invalidate_rect (GTK_WIDGET (pinout)->window, NULL, FALSE);
+        gdk_window_invalidate_rect (
+            gtk_widget_get_window (GTK_WIDGET (pinout)), NULL, FALSE);
       break;
     default:
       G_OBJECT_WARN_INVALID_PROPERTY_ID (object, property_id, pspec);
diff --git a/src/hid/gtk/gui-top-window.c b/src/hid/gtk/gui-top-window.c
index f57dd5c..9c22ea2 100644
--- a/src/hid/gtk/gui-top-window.c
+++ b/src/hid/gtk/gui-top-window.c
@@ -1354,7 +1354,8 @@ ghid_build_pcb_top_window (void)
 
   gtk_widget_show_all (gport->top_window);
   ghid_pack_mode_buttons ();
-  gdk_window_set_back_pixmap (gport->drawing_area->window, NULL, FALSE);
+  gdk_window_set_back_pixmap (gtk_widget_get_window (gport->drawing_area),
+                              NULL, FALSE);
 }
 
 
@@ -1428,28 +1429,30 @@ ghid_interface_set_sensitive (gboolean sensitive)
 static void
 ghid_init_icons (GHidPort * port)
 {
-  XC_clock_source = gdk_bitmap_create_from_data (gport->top_window->window,
+  GdkWindow *window = gtk_widget_get_window (gport->top_window);
+
+  XC_clock_source = gdk_bitmap_create_from_data (window,
 						 (char *) rotateIcon_bits,
 						 rotateIcon_width,
 						 rotateIcon_height);
   XC_clock_mask =
-    gdk_bitmap_create_from_data (gport->top_window->window, (char *) rotateMask_bits,
+    gdk_bitmap_create_from_data (window, (char *) rotateMask_bits,
 				 rotateMask_width, rotateMask_height);
 
-  XC_hand_source = gdk_bitmap_create_from_data (gport->top_window->window,
+  XC_hand_source = gdk_bitmap_create_from_data (window,
 						(char *) handIcon_bits,
 						handIcon_width,
 						handIcon_height);
   XC_hand_mask =
-    gdk_bitmap_create_from_data (gport->top_window->window, (char *) handMask_bits,
+    gdk_bitmap_create_from_data (window, (char *) handMask_bits,
 				 handMask_width, handMask_height);
 
-  XC_lock_source = gdk_bitmap_create_from_data (gport->top_window->window,
+  XC_lock_source = gdk_bitmap_create_from_data (window,
 						(char *) lockIcon_bits,
 						lockIcon_width,
 						lockIcon_height);
   XC_lock_mask =
-    gdk_bitmap_create_from_data (gport->top_window->window, (char *) lockMask_bits,
+    gdk_bitmap_create_from_data (window, (char *) lockMask_bits,
 				 lockMask_width, lockMask_height);
 }
 
diff --git a/src/hid/gtk/gui-utils.c b/src/hid/gtk/gui-utils.c
index e9008c6..db61f77 100644
--- a/src/hid/gtk/gui-utils.c
+++ b/src/hid/gtk/gui-utils.c
@@ -92,7 +92,8 @@ ghid_modifier_keys_state (GdkModifierType * state)
   GHidPort *out = &ghid_port;
 
   if (!state)
-    gdk_window_get_pointer (out->drawing_area->window, NULL, NULL, &mask);
+    gdk_window_get_pointer (gtk_widget_get_window (out->drawing_area),
+                            NULL, NULL, &mask);
   else
     mask = *state;
 
@@ -129,7 +130,8 @@ ghid_button_state (GdkModifierType * state)
   GHidPort *out = &ghid_port;
 
   if (!state)
-    gdk_window_get_pointer (out->drawing_area->window, NULL, NULL, &mask);
+    gdk_window_get_pointer (gtk_widget_get_window (out->drawing_area),
+                            NULL, NULL, &mask);
   else
     mask = *state;
 
@@ -152,7 +154,8 @@ ghid_button_state (GdkModifierType * state)
 void
 ghid_draw_area_update (GHidPort * port, GdkRectangle * rect)
 {
-  gdk_window_invalidate_rect (port->drawing_area->window, rect, FALSE);
+  gdk_window_invalidate_rect (gtk_widget_get_window (port->drawing_area),
+                              rect, FALSE);
 }
